File: LXExtension/UIKit+LXExtension/UIStoryboard+LXExtension.m
//
//  UIStoryboard+LXExtension.m
//
//  Created by 从今以后 on 16/3/5.
//  Copyright © 2016年 从今以后. All rights reserved.
//

#import "UIStoryboard+LXExtension.h"

NS_ASSUME_NONNULL_BEGIN

@implementation UIStoryboard (LXExtension)

#pragma mark - 实例化视图控制器 -

+ (UIViewController *)lx_instantiateInitialViewControllerWithStoryboardName:(NSString *)storyboardName
{
    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:storyboardName bundle:nil];
    UIViewController *viewController = [storyboard instantiateInitialViewController];
    NSAssert(viewController, @"%@ 故事版中未指定初始视图控制器", storyboardName);
    return viewController;
}

+ (UIInputViewController *)lx_instantiateViewControllerWithIdentifier:(NSString *)identifier inStoryBoard:(NSString *)storyboardName
{
    return [[UIStoryboard storyboardWithName:storyboardName bundle:nil] instantiateViewControllerWithIdentifier:identifier];
}

@end

NS_ASSUME_NONNULL_END

File: LXUtilities/UIKitExtensions/Swift/UIStoryboard+LXExtension.swift
//
//  UIStoryboard+LXExtension.swift
//
//  Created by 从今以后 on 2017/10/11.
//  Copyright © 2017年 从今以后. All rights reserved.
//

import UIKit

extension Swifty where Base: UIStoryboard {

    /// 实例化指定故事版中的初始控制器.
    static func instantiateInitialViewController<T: UIViewController>(withStoryboardName name: String) -> T {
        return Base(name: name, bundle: nil).instantiateInitialViewController() as! T
    }

    /// 以类名作为标识符实例化指定故事版中的初始控制器.
    static func instantiateViewController<T: UIViewController>(withStoryboardName name: String, for type: T.Type) -> T {
        return Base(name: name, bundle: nil).instantiateViewController(withIdentifier: String(describing: type)) as! T
    }

    /// 实例化故事版中的初始控制器.
    func instantiateInitialViewController<T: UIViewController>() -> T {
        return base.instantiateInitialViewController() as! T
    }

    /// 以类名作为标识符实例化视图控制器.
    func instantiateViewController<T: UIViewController>(for type: T.Type) -> T {
        return base.instantiateViewController(withIdentifier: String(describing: type)) as! T
    }
}
